### PR TITLE
Bug fixed: #211 setProperty and setArrayIndex set/add cloned nodes

### PR DESCRIPTION
**Pull request by jlolling**
I have made the necessary changes and have also created additional tests.
Best regards
Jan Lolling

### CODE REVIEW

**Comment by jlolling** [comment]
There is an test which fails. I will check that.

**Comment by jlolling** [comment]
All tests are passed now. Sorry.

**Comment by jlolling** [comment]
Thanks for integrating this, it helps me a lot!

File: json-path/src/main/java/com/jayway/jsonpath/spi/json/JacksonJsonNodeJsonProvider.java [hunk 1]
```diff
@@ -237,5 +245,33 @@ private ObjectNode toJsonObject(Object o) {
...
+    	} else if (value instanceof BigDecimal) {
+    		objectNode.put(key.toString(), (BigDecimal) value);
+    	} else if (value instanceof Boolean) {
+    		objectNode.put(key.toString(), (Boolean) value);
+    	} else if (value instanceof byte[]) {
+    		objectNode.put(key.toString(), (byte[]) value);
+    	} else if (value == null) {
+    		objectNode.set(key.toString(), null); // this will create a null-node
+    	} else {
+    		throw new IllegalArgumentException("Cannot handle object type: " + value.getClass().getName());
```
Comment (jochenberger) [inline json-path/src/main/java/com/jayway/jsonpath/spi/json/JacksonJsonNodeJsonProvider.java]: why not `objectNode.set(key.toString(), createJsonElement(value));` here?

**Pull request merged**